src/compiler/tracing.ts

```diff
@@ -88,38 +93,46 @@ namespace ts.tracing {
         Emit = "emit",
     }
 
-    export function begin(phase: Phase, name: string, args: object) {
-        if (!traceFd) {
-            return;
-        }
-        Debug.assert(fs);
+    export type EventData = [phase: Phase, name: string, args?: object];
 
-        performance.mark("beginTracing");
-        fs.writeSync(traceFd, `{"pid":1,"tid":1,"ph":"B","cat":"${phase}","ts":${1000 * timestamp()},"name":"${name}","args":{ "ts": ${JSON.stringify(args)} }},\n`);
-        performance.mark("endTracing");
-        performance.measure("Tracing", "beginTracing", "endTracing");
+    /** Note: `push`/`pop` should be used by default.
+      * `begin`/`end` aren for special cases where we need the data point even if the event never

```

Comment:
I think subsequent lines align with the first star, not the second.

Reply:
Bah.  I don't know how I missed that -- emacs even knows about it...